kolibri/plugins/facility/assets/src/modules/userManagement/handlers.js

```diff
@@ -1,16 +1,14 @@
 import { FacilityUserResource } from 'kolibri.resources';
 import samePageCheckGenerator from 'kolibri.utils.samePageCheckGenerator';
-import { PageNames } from '../../constants';
 import { _userState } from '../mappers';
 
 // An action for setting up the initial state of the app by fetching data from the server
-export function showUserPage(store) {
-  store.dispatch('preparePage', {
-    name: PageNames.USER_MGMT_PAGE,
-  });
+export function showUserPage(store, toRoute) {
+  store.dispatch('preparePage');
+  const facilityId = toRoute.params.facility_id || store.getters.activeFacilityId;
 
   return FacilityUserResource.fetchCollection({
-    getParams: { member_of: store.getters.currentFacilityId },
+    getParams: { member_of: facilityId },
```

Comment:
This is failing when an admin (not a superuser) is logged in the facility where it belongs.
This is the request 
http://localhost:8000/api/auth/facilityuser/?member_of=4ded44334b426f2db29e481ea5e00354&1586360084013=1586360084013

An this is the traceback:
```
    MultipleObjectsReturned at /api/auth/facilityuser/
    get() returned more than one Facility -- it returned 2!
    Request Method: GET
    Request URL: http://localhost:8000/api/auth/facilityuser/?member_of=4ded44334b426f2db29e481ea5e00354&1586360084013=1586360084013
    Django Version: 1.11.28
    Python Executable: /datos/le/mio/kolibri/venv/bin/python3
    Python Version: 3.7.0
    Python Path: ['/datos/le/mio/kolibri/kolibri/dist', '/datos/le/mio/kolibri/venv/bin', '/usr/local/lib/python37.zip', '/usr/local/lib/python3.7', '/usr/local/lib/python3.7/lib-dynload', '/datos/le/mio/kolibri/venv/lib/python3.7/site-packages', '/datos/le/mio/kolibri']
    Server time: Wed, 8 Apr 2020 17:34:44 +0200
    Installed Applications:
    ['kolibri.core',
     'django.contrib.admin',
     'django.contrib.auth',
     'django.contrib.contenttypes',
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'django_filters',
     'kolibri.core.auth.apps.KolibriAuthConfig',
     'kolibri.core.content',
     'kolibri.core.logger',
     'kolibri.core.notifications.apps.KolibriNotificationsConfig',
     'kolibri.core.tasks.apps.KolibriTasksConfig',
     'kolibri.core.deviceadmin',
     'kolibri.core.webpack',
     'kolibri.core.exams',
     'kolibri.core.device',
     'kolibri.core.discovery',
     'kolibri.core.lessons',
     'kolibri.core.analytics',
     'rest_framework',
     'django_js_reverse',
     'jsonfield',
     'morango',
     <AppConfig: kolibri_exercise_perseus_plugin>,
     <AppConfig: kolibri.plugins.style_guide>,
     <AppConfig: kolibri.plugins.epub_viewer>,
     <AppConfig: kolibri.plugins.setup_wizard>,
     <AppConfig: kolibri.plugins.media_player>,
     <AppConfig: kolibri.plugins.html5_viewer>,
     <AppConfig: kolibri.plugins.slideshow_viewer>,
     <AppConfig: kolibri.plugins.coach>,
     <AppConfig: kolibri.plugins.device>,
     <AppConfig: kolibri.plugins.default_theme>,
     <AppConfig: kolibri.plugins.h5p_viewer>,
     <AppConfig: kolibri.plugins.facility>,
     <AppConfig: kolibri.plugins.user>,
     <AppConfig: kolibri.plugins.learn>,
     <AppConfig: kolibri.plugins.pdf_viewer>,
     'rest_framework_swagger']
    Installed Middleware:
    ['django.middleware.cache.UpdateCacheMiddleware',
     'kolibri.core.analytics.middleware.MetricsMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'kolibri.core.device.middleware.KolibriLocaleMiddleware',
     'django.middleware.common.CommonMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'kolibri.core.auth.middleware.CustomAuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
     'django.middleware.security.SecurityMiddleware',
     'django.middleware.cache.FetchFromCacheMiddleware',
     'kolibri.core.webpack.middleware.WebpackErrorHandler']
    Traceback:  
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
      41.             response = get_response(request)
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
      187.                 response = self.process_exception_by_middleware(e, request)
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
      185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/django/views/decorators/csrf.py" in wrapped_view
      58.         return view_func(*args, **kwargs)
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/rest_framework/viewsets.py" in view
      116.             return self.dispatch(request, *args, **kwargs)
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/rest_framework/views.py" in dispatch
      495.             response = self.handle_exception(exc)
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/rest_framework/views.py" in handle_exception
      455.             self.raise_uncaught_exception(exc)
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/rest_framework/views.py" in dispatch
      492.             response = handler(request, *args, **kwargs)
    File "/datos/le/mio/kolibri/kolibri/core/api.py" in list
      103.         queryset = self.filter_queryset(self.prefetch_queryset(self.get_queryset()))
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/rest_framework/generics.py" in filter_queryset
      152.             queryset = backend().filter_queryset(self.request, queryset, self)
    File "/datos/le/mio/kolibri/kolibri/core/auth/api.py" in filter_queryset
      72.             return request.user.filter_readable(queryset)
    File "/datos/le/mio/kolibri/kolibri/core/auth/models.py" in filter_readable
      897.                 self, queryset
    File "/datos/le/mio/kolibri/kolibri/core/auth/permissions/base.py" in readable_by_user_filter
      232.                 user, queryset
    File "/datos/le/mio/kolibri/kolibri/core/auth/permissions/base.py" in readable_by_user_filter
      232.                 user, queryset
    File "/datos/le/mio/kolibri/kolibri/core/auth/permissions/general.py" in readable_by_user_filter
      182.         if _user_is_admin_for_own_facility(user):
    File "/datos/le/mio/kolibri/kolibri/core/auth/permissions/general.py" in _user_is_admin_for_own_facility
      157.     facility = Facility.objects.get(dataset=user.dataset)
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/django/db/models/manager.py" in manager_method
      85.                 return getattr(self.get_queryset(), name)(*args, **kwargs)
    File "/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/django/db/models/query.py" in get
      384.             (self.model._meta.object_name, num)
    Exception Type: MultipleObjectsReturned at /api/auth/facilityuser/
    Exception Value: get() returned more than one Facility -- it returned 2!
    Request information:
    USER: "jose"@"uno"
    GET:
    member_of = '4ded44334b426f2db29e481ea5e00354'
    1586360084013 = '1586360084013'
    POST: No POST data
    FILES: No FILES data
    COOKIES:
    kolibri_csrftoken = 'A2z33vGenPuMKNGGTCepgcjlHddrfHbJA0fv1vjomwYSwfJjrs62KcyTLKkqoz5f'
    kolibri = 'ux0yureefnrpnohfuaqi3omsc3fak96y'
    META:
    CINNAMON_VERSION = '4.2.4'
    COLORTERM = 'truecolor'
    CONTENT_LENGTH = ''
    CONTENT_TYPE = 'application/json'
    CSRF_COOKIE = 'A2z33vGenPuMKNGGTCepgcjlHddrfHbJA0fv1vjomwYSwfJjrs62KcyTLKkqoz5f'
    DBUS_SESSION_BUS_ADDRESS = 'unix:path=/run/user/1000/bus'
    DESKTOP_SESSION = 'cinnamon'
    DISPLAY = ':0'
    DJANGO_SETTINGS_MODULE = 'kolibri.deployment.default.settings.dev'
    GATEWAY_INTERFACE = 'CGI/1.1'
    GDMSESSION = 'cinnamon'
    GIO_LAUNCHED_DESKTOP_FILE = '/usr/share/applications/terminator.desktop'
    GIO_LAUNCHED_DESKTOP_FILE_PID = '4021'
    GJS_DEBUG_OUTPUT = 'stderr'
    GJS_DEBUG_TOPICS = 'JS ERROR;JS LOG'
    GNOME_DESKTOP_SESSION_ID = 'this-is-deprecated'
    GPG_AGENT_INFO = '/run/user/1000/gnupg/S.gpg-agent:0:1'
    GTK_MODULES = 'gail:atk-bridge'
    GTK_OVERLAY_SCROLLING = '1'
    HOME = '/home/jose'
    HTTP_ACCEPT = 'application/json, application/json;q=0.8, text/plain;q=0.5, */*;q=0.2'
    HTTP_ACCEPT_ENCODING = 'gzip, deflate'
    HTTP_ACCEPT_LANGUAGE = 'es-ES,es;q=0.8,en-US;q=0.5,en;q=0.3'
    HTTP_CONNECTION = 'keep-alive'
    HTTP_COOKIE = 'kolibri_csrftoken=A2z33vGenPuMKNGGTCepgcjlHddrfHbJA0fv1vjomwYSwfJjrs62KcyTLKkqoz5f; kolibri=ux0yureefnrpnohfuaqi3omsc3fak96y'
    HTTP_DNT = '1'
    HTTP_HOST = 'localhost:8000'
    HTTP_REFERER = 'http://localhost:8000/es-es/facility/'
    HTTP_USER_AGENT = 'Mozilla/5.0 (X11; Linux x86_64; rv:76.0) Gecko/20100101 Firefox/76.0'
    HTTP_X_CSRFTOKEN = 'A2z33vGenPuMKNGGTCepgcjlHddrfHbJA0fv1vjomwYSwfJjrs62KcyTLKkqoz5f'
    INIT_CWD = '/datos/le/mio/kolibri'
    KOLIBRI_HOME = '/home/jose/.kolibri'
    KOLIBRI_RUN_MODE = 'jose'
    LANG = 'es_ES.UTF-8'
    LOGNAME = 'jose'
    LS_COLORS = 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:'
    NODE = '/home/jose/.nvm/versions/node/v10.19.0/bin/node'
    NVM_BIN = '/home/jose/.nvm/versions/node/v10.19.0/bin'
    NVM_CD_FLAGS = ''
    NVM_DIR = '/home/jose/.nvm'
    OLDPWD = '/datos/le/mio'
    PATH = '/home/jose/.nvm/versions/node/v10.19.0/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/datos/le/mio/kolibri/node_modules/.bin:/tmp/yarn--1586335704636-0.860555689304932:/datos/le/mio/kolibri/node_modules/.bin:/home/jose/.config/yarn/link/node_modules/.bin:/datos/le/mio/kolibri/node_modules/.bin:/home/jose/.yarn/bin:/home/jose/.nvm/versions/node/v10.19.0/libexec/lib/node_modules/npm/bin/node-gyp-bin:/home/jose/.nvm/versions/node/v10.19.0/lib/node_modules/npm/bin/node-gyp-bin:/home/jose/.nvm/versions/node/v10.19.0/bin/node_modules/npm/bin/node-gyp-bin:/datos/le/mio/kolibri/venv/bin:/home/jose/.nvm/versions/node/v10.19.0/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/usr/local/sbin:/usr/sbin:/sbin'
    PATH_INFO = '/api/auth/facilityuser/'
    PROMPT_COMMAND = 'if [ "$(id -u)" -ne 0 ]; then echo "$(date "+%Y-%m-%d.%H:%M:%S") $(pwd) $(history 1)" >> ~/.logs/bash-history-$(date "+%Y-%m-%d").log; fi'
    PS1 = '(venv) \\[\\e]0;\\u@\\h: \\w\\a\\]${debian_chroot:+($debian_chroot)}\\u@\\h:\\w\\$ '
    PWD = '/datos/le/mio/kolibri'
    QT_ACCESSIBILITY = '1'
    QT_LINUX_ACCESSIBILITY_ALWAYS_ON = '1'
    QT_QPA_PLATFORMTHEME = 'gtk2'
    QUERY_STRING = 'member_of=4ded44334b426f2db29e481ea5e00354&1586360084013=1586360084013'
    REMOTE_ADDR = '127.0.0.1'
    REMOTE_HOST = ''
    REQUEST_METHOD = 'GET'
    RUN_MAIN = 'true'
    SCRIPT_NAME = ''
    SERVER_NAME = 'casa-pro'
    SERVER_PORT = '8000'
    SERVER_PROTOCOL = 'HTTP/1.1'
    SERVER_SOFTWARE = 'WSGIServer/0.2'
    SESSION_MANAGER = 'local/casa-pro:@/tmp/.ICE-unix/892,unix/casa-pro:/tmp/.ICE-unix/892'
    SHELL = '/bin/bash'
    SHLVL = '1'
    SSH_AGENT_PID = '997'
    SSH_AUTH_SOCK = '/run/user/1000/keyring/ssh'
    TERM = 'xterm-256color'
    TERMINATOR_DBUS_NAME = 'net.tenshu.Terminator20x1a6021154d881c'
    TERMINATOR_DBUS_PATH = '/net/tenshu/Terminator2'
    TERMINATOR_UUID = 'urn:uuid:6ff006df-0732-43a9-a5c1-275539f0461d'
    TZ = 'Europe/Madrid'
    USER = 'jose'
    VIRTUAL_ENV = '/datos/le/mio/kolibri/venv'
    VTE_VERSION = '4601'
    XAUTHORITY = '/home/jose/.Xauthority'
    XDG_CURRENT_DESKTOP = 'X-Cinnamon'
    XDG_DATA_DIRS = '/usr/share/gnome:/home/jose/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share:/usr/share'
    XDG_GREETER_DATA_DIR = '/var/lib/lightdm/data/jose'
    XDG_RUNTIME_DIR = '/run/user/1000'
    XDG_SEAT = 'seat0'
    XDG_SEAT_PATH = '/org/freedesktop/DisplayManager/Seat0'
    XDG_SESSION_DESKTOP = 'cinnamon'
    XDG_SESSION_ID = '1'
    XDG_SESSION_PATH = '/org/freedesktop/DisplayManager/Session0'
    XDG_SESSION_TYPE = 'x11'
    XDG_VTNR = '7'
    YARN_WRAP_OUTPUT = 'false'
    _ = '/usr/bin/yarn'
    npm_config_access = ''
    npm_config_allow_same_version = ''
    npm_config_also = ''
    npm_config_always_auth = ''
    npm_config_argv = '{"remain":[],"cooked":["run","django-devserver-no-update"],"original":["run","django-devserver-no-update"]}'
    npm_config_audit = 'true'
    npm_config_audit_level = 'low'
    npm_config_auth_type = 'legacy'
    npm_config_before = ''
    npm_config_bin_links = 'true'
    npm_config_browser = ''
    npm_config_ca = ''
    npm_config_cache = '/home/jose/.npm'
    npm_config_cache_lock_retries = '10'
    npm_config_cache_lock_stale = '60000'
    npm_config_cache_lock_wait = '10000'
    npm_config_cache_max = 'Infinity'
    npm_config_cache_min = '10'
    npm_config_cafile = ''
    npm_config_cert = ''
    npm_config_cidr = ''
    npm_config_color = 'true'
    npm_config_commit_hooks = 'true'
    npm_config_depth = 'Infinity'
    npm_config_description = 'true'
    npm_config_dev = ''
    npm_config_dry_run = ''
    npm_config_editor = 'vi'
    npm_config_engine_strict = ''
    npm_config_fetch_retries = '2'
    npm_config_fetch_retry_factor = '10'
    npm_config_fetch_retry_maxtimeout = '60000'
    npm_config_fetch_retry_mintimeout = '10000'
    npm_config_force = ''
    npm_config_format_package_lock = 'true'
    npm_config_fund = 'true'
    npm_config_git = 'git'
    npm_config_git_tag_version = 'true'
    npm_config_global = ''
    npm_config_global_style = ''
    npm_config_globalconfig = '/home/jose/.nvm/versions/node/v10.19.0/etc/npmrc'
    npm_config_globalignorefile = '/home/jose/.nvm/versions/node/v10.19.0/etc/npmignore'
    npm_config_group = '1000'
    npm_config_ham_it_up = ''
    npm_config_heading = 'npm'
    npm_config_https_proxy = ''
    npm_config_if_present = ''
    npm_config_ignore_prepublish = ''
    npm_config_ignore_scripts = ''
    npm_config_init_author_email = ''
    npm_config_init_author_name = ''
    npm_config_init_author_url = ''
    npm_config_init_license = 'MIT'
    npm_config_init_module = '/home/jose/.npm-init.js'
    npm_config_init_version = '1.0.0'
    npm_config_json = ''
    npm_config_key = ''
    npm_config_legacy_bundling = ''
    npm_config_link = ''
    npm_config_local_address = ''
    npm_config_loglevel = 'notice'
    npm_config_logs_max = '10'
    npm_config_long = ''
    npm_config_maxsockets = '50'
    npm_config_message = '%s'
    npm_config_metrics_registry = 'https://registry.yarnpkg.com/'
    npm_config_node_gyp = '/home/jose/.nvm/versions/node/v10.19.0/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js'
    npm_config_node_options = ''
    npm_config_node_version = '10.19.0'
    npm_config_noproxy = ''
    npm_config_offline = ''
    npm_config_onload_script = ''
    npm_config_only = ''
    npm_config_optional = 'true'
    npm_config_otp = ''
    npm_config_package_lock = 'true'
    npm_config_package_lock_only = ''
    npm_config_parseable = ''
    npm_config_prefer_offline = ''
    npm_config_prefer_online = ''
    npm_config_prefix = '/home/jose/.nvm/versions/node/v10.19.0'
    npm_config_preid = ''
    npm_config_production = ''
    npm_config_progress = 'true'
    npm_config_proxy = ''
    npm_config_read_only = ''
    npm_config_rebuild_bundle = 'true'
    npm_config_registry = 'https://registry.yarnpkg.com/'
    npm_config_rollback = 'true'
    npm_config_save = 'true'
    npm_config_save_bundle = ''
    npm_config_save_dev = ''
    npm_config_save_exact = ''
    npm_config_save_optional = ''
    npm_config_save_prefix = '^'
    npm_config_save_prod = ''
    npm_config_scope = ''
    npm_config_script_shell = ''
    npm_config_scripts_prepend_node_path = 'warn-only'
    npm_config_searchexclude = ''
    npm_config_searchlimit = '20'
    npm_config_searchopts = ''
    npm_config_searchstaleness = '900'
    npm_config_send_metrics = ''
    npm_config_shell = '/bin/bash'
    npm_config_shrinkwrap = 'true'
    npm_config_sign_git_commit = ''
    npm_config_sign_git_tag = ''
    npm_config_sso_poll_frequency = '500'
    npm_config_sso_type = 'oauth'
    npm_config_strict_ssl = 'true'
    npm_config_tag = 'latest'
    npm_config_tag_version_prefix = 'v'
    npm_config_timing = ''
    npm_config_tmp = '/tmp'
    npm_config_umask = '0022'
    npm_config_unicode = 'true'
    npm_config_unsafe_perm = 'true'
    npm_config_update_notifier = 'true'
    npm_config_usage = ''
    npm_config_user = '1000'
    npm_config_user_agent = 'yarn/1.22.4 npm/? node/v10.19.0 linux x64'
    npm_config_userconfig = '/home/jose/.npmrc'
    npm_config_version = ''
    npm_config_version_commit_hooks = 'true'
    npm_config_version_git_message = 'v%s'
    npm_config_version_git_tag = 'true'
    npm_config_version_tag_prefix = 'v'
    npm_config_versions = ''
    npm_config_viewer = 'man'
    npm_execpath = '/home/jose/.nvm/versions/node/v10.19.0/lib/node_modules/npm/bin/npm-cli.js'
    npm_lifecycle_event = 'django-devserver-no-update'
    npm_lifecycle_script = 'kolibri manage --debug --skip-update runserver --settings=kolibri.deployment.default.settings.dev "0.0.0.0:8000" '
    npm_node_execpath = '/home/jose/.nvm/versions/node/v10.19.0/bin/node'
    npm_package_author_name = 'Learning Equality'
    npm_package_bugs_url = 'https://github.com/learningequality/kolibri/issues'
    npm_package_dependencies_espree = '5.0.1'
    npm_package_dependencies_fsevents = '*'
    npm_package_description = 'Development utilities for Kolibri'
    npm_package_devDependencies__types_jest = '^24.0.12'
    npm_package_devDependencies_black_fmt = 'git+https://github.com/learningequality/black-fmt.git#v0.1.3'
    npm_package_devDependencies_kolibri_tools = '0.13.0-dev.10'
    npm_package_devDependencies_yarn_run_all = '^3.1.1'
    npm_package_engineStrict = 'true'
    npm_package_engines_node = '10.x'
    npm_package_engines_yarn = '>= 1.12.3'
    npm_package_gitHead = '5ea49b2410b8bb31e0feed25932179f41ac1ff7b'
    npm_package_homepage = 'https://github.com/learningequality/kolibri#readme'
    npm_package_license = 'MIT'
    npm_package_name = 'kolibri-root'
    npm_package_optionalDependencies_fsevents = '*'
    npm_package_private = 'true'
    npm_package_readmeFilename = 'README.md'
    npm_package_repository_type = 'git'
    npm_package_repository_url = 'git+https://github.com/learningequality/kolibri.git'
    npm_package_scripts_build = 'kolibri-tools build prod --file ./build_tools/build_plugins.txt && yarn run hashi-build'
    npm_package_scripts_build_kolibri_tools = 'yarn workspace kolibri-tools run build-kolibri-tools'
    npm_package_scripts_bundle_stats = 'kolibri-tools build stats --file ./build_tools/build_plugins.txt'
    npm_package_scripts_clean = 'kolibri-tools build clean --file ./build_tools/build_plugins.txt'
    npm_package_scripts_combineprofiles = 'node ./packages/kolibri-tools/lib/combineStringProfiles.js ./kolibri/locale/en/LC_MESSAGES/profiles/'
    npm_package_scripts_coverage = 'yarn run test-jest-cov'
    npm_package_scripts_devserver = 'npm-run-all --parallel django-devserver-no-update lint-frontend:watch:format hashi-dev watch services'
    npm_package_scripts_devserver_hot = 'npm-run-all --parallel django-devserver-no-update lint-frontend:watch:format hashi-dev watch-hot services'
    npm_package_scripts_devserver_hot_warn = 'npm-run-all --parallel django-devserver-no-update lint-frontend:watch hashi-dev watch-hot services'
    npm_package_scripts_devserver_warn = 'npm-run-all --parallel django-devserver-no-update lint-frontend:watch hashi-dev watch services'
    npm_package_scripts_django_devserver = 'kolibri manage --debug runserver --settings=kolibri.deployment.default.settings.dev "0.0.0.0:8000" '
    npm_package_scripts_django_devserver_no_update = 'kolibri manage --debug --skip-update runserver --settings=kolibri.deployment.default.settings.dev "0.0.0.0:8000" '
    npm_package_scripts_fmt_backend = 'yarn run black-fmt'
    npm_package_scripts_fmt_backend_check = 'yarn run black-fmt --check'
    npm_package_scripts_fmt_backend_watch = 'yarn run black-fmt --watch'
    npm_package_scripts_hashi_build = 'yarn workspace hashi run build'
    npm_package_scripts_hashi_dev = 'yarn workspace hashi run dev'
    npm_package_scripts_lint_frontend = "kolibri-tools lint '{kolibri*/**/assets,packages}/**/*.{js,vue,scss,less,css}' --ignore '**/node_modules/**','**/static/**','**/packages/kolibri-core-for-export/**'"
    npm_package_scripts_lint_frontend_format = 'yarn run lint-frontend --write'
    npm_package_scripts_lint_frontend_watch = 'yarn run lint-frontend --monitor'
    npm_package_scripts_lint_frontend_watch_format = 'yarn run lint-frontend --monitor --write'
    npm_package_scripts_makemessages = 'kolibri-tools build i18n --file ./build_tools/build_plugins.txt'
    npm_package_scripts_preinstall = 'node ./packages/kolibri-tools/lib/npm_deprecation_warning.js'
    npm_package_scripts_publish_packages = 'node ./packages/publish.js'
    npm_package_scripts_services = 'kolibri services --foreground --port=8000'
    npm_package_scripts_test = 'yarn run test-jest --watch'
    npm_package_scripts_test_jest = 'kolibri-tools test --config ./jest.conf.js'
    npm_package_scripts_test_jest_cov = 'yarn run test-jest --coverage'
    npm_package_scripts_transfercontext = 'node ./packages/kolibri-tools/lib/i18nSyncContext.js run && yarn lint-frontend:format'
    npm_package_scripts_watch = 'kolibri-tools build dev --file ./build_tools/build_plugins.txt'
    npm_package_scripts_watch_hot = 'yarn run watch --hot'
    npm_package_version = '0.12.0-dev.1'
    npm_package_workspaces_0 = 'packages/*'
    npm_package_workspaces_1 = 'kolibri/core'
    npm_package_workspaces_2 = 'kolibri/plugins/*'
    wsgi.errors = <_io.TextIOWrapper name='<stderr>' mode='w' encoding='UTF-8'>
    wsgi.file_wrapper = ''
    wsgi.input = <_io.BufferedReader name=4>
    wsgi.multiprocess = False
    wsgi.multithread = True
    wsgi.run_once = False
    wsgi.url_scheme = 'http'
    wsgi.version = '(1, 0)'
    Settings:
    Using settings module kolibri.deployment.default.settings.dev
    ABSOLUTE_URL_OVERRIDES = {}
    ADMINS = []
    ALLOWED_HOSTS = ['*']
    APPEND_SLASH = True
    AUTHENTICATION_BACKENDS = ['kolibri.core.auth.backends.FacilityUserBackend']
    AUTH_ANONYMOUS_USER_MODEL = 'kolibriauth.KolibriAnonymousUser'
    AUTH_PASSWORD_VALIDATORS = '********************'
    AUTH_USER_MODEL = 'kolibriauth.FacilityUser'
    BASE_DIR = '/datos/le/mio/kolibri'
    CACHES = {'default': {'BACKEND': 'django.core.cache.backends.locmem.LocMemCache'}, 'built_files': {'BACKEND': 'django.core.cache.backends.locmem.LocMemCache'}, 'process_cache': {'BACKEND': 'django.core.cache.backends.locmem.LocMemCache'}}
    CACHE_MIDDLEWARE_ALIAS = 'default'
    CACHE_MIDDLEWARE_KEY_PREFIX = '********************'
    CACHE_MIDDLEWARE_SECONDS = 0
    CSRF_COOKIE_AGE = 31449600
    CSRF_COOKIE_DOMAIN = None
    CSRF_COOKIE_HTTPONLY = False
    CSRF_COOKIE_NAME = 'kolibri_csrftoken'
    CSRF_COOKIE_PATH = '/'
    CSRF_COOKIE_SECURE = False
    CSRF_FAILURE_VIEW = 'django.views.csrf.csrf_failure'
    CSRF_HEADER_NAME = 'HTTP_X_CSRFTOKEN'
    CSRF_TRUSTED_ORIGINS = []
    CSRF_USE_SESSIONS = False
    DATABASES = {'default': {'ENGINE': 'django.db.backends.sqlite3', 'NAME': '/home/jose/.kolibri/db.sqlite3', 'OPTIONS': {'timeout': 100}, 'ATOMIC_REQUESTS': False, 'AUTOCOMMIT': True, 'CONN_MAX_AGE': 0, 'TIME_ZONE': None, 'USER': '', 'PASSWORD': '********************', 'HOST': '', 'PORT': '', 'TEST': {'CHARSET': None, 'COLLATION': None, 'NAME': None, 'MIRROR': None}}, 'notifications_db': {'ENGINE': 'django.db.backends.sqlite3', 'NAME': '/home/jose/.kolibri/notifications.sqlite3', 'OPTIONS': {'timeout': 100}, 'ATOMIC_REQUESTS': False, 'AUTOCOMMIT': True, 'CONN_MAX_AGE': 0, 'TIME_ZONE': None, 'USER': '', 'PASSWORD': '********************', 'HOST': '', 'PORT': '', 'TEST': {'CHARSET': None, 'COLLATION': None, 'NAME': None, 'MIRROR': None}}}
    DATABASE_ROUTERS = "('kolibri.core.notifications.models.NotificationsRouter',)"
    DATA_UPLOAD_MAX_MEMORY_SIZE = 2621440
    DATA_UPLOAD_MAX_NUMBER_FIELDS = 1000
    DATETIME_FORMAT = 'N j, Y, P'
    DATETIME_INPUT_FORMATS = ['%Y-%m-%d %H:%M:%S', '%Y-%m-%d %H:%M:%S.%f', '%Y-%m-%d %H:%M', '%Y-%m-%d', '%m/%d/%Y %H:%M:%S', '%m/%d/%Y %H:%M:%S.%f', '%m/%d/%Y %H:%M', '%m/%d/%Y', '%m/%d/%y %H:%M:%S', '%m/%d/%y %H:%M:%S.%f', '%m/%d/%y %H:%M', '%m/%d/%y']
    DATE_FORMAT = 'N j, Y'
    DATE_INPUT_FORMATS = ['%Y-%m-%d', '%m/%d/%Y', '%m/%d/%y', '%b %d %Y', '%b %d, %Y', '%d %b %Y', '%d %b, %Y', '%B %d %Y', '%B %d, %Y', '%d %B %Y', '%d %B, %Y']
    DEBUG = True
    DEBUG_PROPAGATE_EXCEPTIONS = False
    DECIMAL_SEPARATOR = '.'
    DEFAULT_CHARSET = 'utf-8'
    DEFAULT_CONTENT_TYPE = 'text/html'
    DEFAULT_EXCEPTION_REPORTER_FILTER = 'django.views.debug.SafeExceptionReporterFilter'
    DEFAULT_FILE_STORAGE = 'django.core.files.storage.FileSystemStorage'
    DEFAULT_FROM_EMAIL = 'webmaster@localhost'
    DEFAULT_INDEX_TABLESPACE = ''
    DEFAULT_TABLESPACE = ''
    DEVELOPER_MODE = True
    DISALLOWED_USER_AGENTS = []
    EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
    EMAIL_HOST = 'localhost'
    EMAIL_HOST_PASSWORD = '********************'
    EMAIL_HOST_USER = ''
    EMAIL_PORT = 25
    EMAIL_SSL_CERTFILE = None
    EMAIL_SSL_KEYFILE = '********************'
    EMAIL_SUBJECT_PREFIX = '[Django] '
    EMAIL_TIMEOUT = None
    EMAIL_USE_LOCALTIME = False
    EMAIL_USE_SSL = False
    EMAIL_USE_TLS = False
    ENABLE_DATA_BOOTSTRAPPING = True
    EXTRA_LANG_INFO = {'ff-cm': {'bidi': False, 'code': 'ff-cm', 'name': 'Fulfulde (Cameroon)', 'name_local': 'Fulfulde Mbororoore', 'name_translated': 'Fulfulde (Cameroon)'}, 'es-419': {'bidi': False, 'code': 'es-419', 'name': 'Spanish (Latin America)', 'name_local': 'Español', 'name_translated': 'Spanish (Latin America)'}, 'es-es': {'bidi': False, 'code': 'es-es', 'name': 'Spanish (Spain)', 'name_local': 'Español (España)', 'name_translated': 'Spanish (Spain)'}, 'fr-ht': {'bidi': False, 'code': 'fr-ht', 'name': 'Haitian Creole', 'name_local': 'Kreyòl ayisyen'}, 'gu-in': {'bidi': False, 'code': 'gu-in', 'name': 'Gujarati', 'name_local': 'ગુજરાતી', 'name_translated': 'Gujarati'}, 'nyn': {'bidi': False, 'code': 'nyn', 'name': 'Chichewa, Chewa, Nyanja', 'name_local': 'Chinyanja', 'name_translated': 'Chichewa, Chewa, Nyanja'}, 'yo': {'bidi': False, 'code': 'yo', 'name': 'Yoruba', 'name_local': 'Yorùbá', 'name_translated': 'Yoruba'}, 'zu': {'bidi': False, 'code': 'zu', 'name': 'Zulu', 'name_local': 'isiZulu'}}
    FILE_CHARSET = 'utf-8'
    FILE_UPLOAD_DIRECTORY_PERMISSIONS = None
    FILE_UPLOAD_HANDLERS = ['django.core.files.uploadhandler.MemoryFileUploadHandler', 'django.core.files.uploadhandler.TemporaryFileUploadHandler']
    FILE_UPLOAD_MAX_MEMORY_SIZE = 2621440
    FILE_UPLOAD_PERMISSIONS = None
    FILE_UPLOAD_TEMP_DIR = None
    FIRST_DAY_OF_WEEK = 0
    FIXTURE_DIRS = []
    FORCE_SCRIPT_NAME = None
    FORMAT_MODULE_PATH = None
    FORM_RENDERER = 'django.forms.renderers.DjangoTemplates'
    IGNORABLE_404_URLS = []
    INSTALLED_APPS = ['kolibri.core', 'django.contrib.admin', 'django.contrib.auth', 'django.contrib.contenttypes', 'django.contrib.sessions', 'django.contrib.messages', 'django.contrib.staticfiles', 'django_filters', 'kolibri.core.auth.apps.KolibriAuthConfig', 'kolibri.core.content', 'kolibri.core.logger', 'kolibri.core.notifications.apps.KolibriNotificationsConfig', 'kolibri.core.tasks.apps.KolibriTasksConfig', 'kolibri.core.deviceadmin', 'kolibri.core.webpack', 'kolibri.core.exams', 'kolibri.core.device', 'kolibri.core.discovery', 'kolibri.core.lessons', 'kolibri.core.analytics', 'rest_framework', 'django_js_reverse', 'jsonfield', 'morango', <AppConfig: kolibri_exercise_perseus_plugin>, <AppConfig: kolibri.plugins.style_guide>, <AppConfig: kolibri.plugins.epub_viewer>, <AppConfig: kolibri.plugins.setup_wizard>, <AppConfig: kolibri.plugins.media_player>, <AppConfig: kolibri.plugins.html5_viewer>, <AppConfig: kolibri.plugins.slideshow_viewer>, <AppConfig: kolibri.plugins.coach>, <AppConfig: kolibri.plugins.device>, <AppConfig: kolibri.plugins.default_theme>, <AppConfig: kolibri.plugins.h5p_viewer>, <AppConfig: kolibri.plugins.facility>, <AppConfig: kolibri.plugins.user>, <AppConfig: kolibri.plugins.learn>, <AppConfig: kolibri.plugins.pdf_viewer>, 'rest_framework_swagger']
    INTERNAL_IPS = ['127.0.0.1']
    JS_REVERSE_EXCLUDE_NAMESPACES = ['admin']
    KOLIBRI_MODULE_PATH = '/datos/le/mio/kolibri/kolibri'
    LANGUAGES = [('ar', 'العَرَبِيَّة\u200e\u200e'), ('bg-bg', 'Български'), ('bn-bd', 'বাংলা'), ('en', 'English'), ('es-es', 'Español (España)'), ('es-419', 'Español'), ('fa', 'فارسی'), ('fr-fr', 'Français'), ('ff-cm', 'Fulfulde Mbororoore'), ('gu-in', 'ગુજરાતી'), ('hi-in', 'हिंदी (भारत)'), ('ko', '한국어'), ('mr', 'मराठी'), ('my', 'ဗမာစာ'), ('nyn', 'Chinyanja'), ('pt-br', 'Português (Brasil)'), ('sw-tz', 'Kiswahili'), ('te', 'తెలుగు'), ('ur-pk', 'اُردو (پاکستان)\u200f'), ('vi', 'Tiếng Việt'), ('yo', 'Yorùbá')]
    LANGUAGES_BIDI = ['he', 'ar', 'fa', 'ur']
    LANGUAGE_CODE = 'en'
    LANGUAGE_COOKIE_AGE = None
    LANGUAGE_COOKIE_DOMAIN = None
    LANGUAGE_COOKIE_NAME = 'django_language'
    LANGUAGE_COOKIE_PATH = '/'
    LOCALE_PATHS = "('/datos/le/mio/kolibri/kolibri/locale', '/datos/le/mio/kolibri/venv/lib/python3.7/site-packages/kolibri_exercise_perseus_plugin/locale')"
    LOGGING = {'version': 1, 'disable_existing_loggers': False, 'formatters': {'verbose': {'format': '%(levelname)s %(asctime)s %(module)s %(process)d %(thread)d %(message)s'}, 'simple': {'format': '%(levelname)s %(message)s'}, 'simple_date': {'format': '%(levelname)s %(asctime)s %(module)s %(message)s'}, 'simple_date_file': {'()': 'kolibri.utils.logger.KolibriLogFileFormatter', 'format': '%(levelname)s %(asctime)s %(module)s %(message)s'}, 'color': {'()': 'colorlog.ColoredFormatter', 'format': '%(log_color)s%(levelname)-8s %(message)s', 'log_colors': {'DEBUG': 'bold_black', 'INFO': 'white', 'WARNING': 'yellow', 'ERROR': 'red', 'CRITICAL': 'bold_red'}}}, 'handlers': {'console': {'level': 'INFO', 'class': 'logging.StreamHandler', 'formatter': 'color'}, 'file': {'level': 'INFO', 'filters': [], 'class': 'kolibri.utils.logger.KolibriTimedRotatingFileHandler', 'filename': '/home/jose/.kolibri/logs/kolibri.txt', 'formatter': 'simple_date_file', 'when': 'midnight', 'backupCount': 30}, 'mail_admins': {'level': 'ERROR', 'class': 'django.utils.log.AdminEmailHandler', 'filters': ['require_debug_false']}, 'request_debug': {'level': 'ERROR', 'class': 'logging.StreamHandler', 'formatter': 'color', 'filters': ['require_debug_true']}, 'file_debug': {'level': 'DEBUG', 'filters': ['require_debug_true'], 'class': 'logging.FileHandler', 'filename': '/home/jose/.kolibri/logs/debug.txt', 'formatter': 'simple_date'}}, 'loggers': {'kolibri': {'handlers': ['console', 'mail_admins', 'file', 'file_debug'], 'level': 'INFO', 'propagate': False}, 'iceqube': {'handlers': ['file', 'console'], 'level': 'INFO', 'propagate': False}, 'morango': {'handlers': ['file', 'console'], 'level': 'INFO', 'propagate': False}, 'cherrypy.access': {'handlers': ['file', 'console'], 'level': 'INFO', 'propagate': False}, 'cherrypy.error': {'handlers': ['file', 'console'], 'level': 'INFO', 'propagate': False}, 'cherrypy': {'handlers': ['file', 'console'], 'level': 'INFO', 'propagate': False}, 'django': {'handlers': ['console', 'file'], 'propagate': False}, 'django.request': {'handlers': ['mail_admins', 'file', 'request_debug'], 'level': 'ERROR', 'propagate': False}}, 'filters': {'require_debug_true': {'()': 'django.utils.log.RequireDebugTrue'}, 'require_debug_false': {'()': 'django.utils.log.RequireDebugFalse'}}}
    LOGGING_CONFIG = 'logging.config.dictConfig'
    LOGIN_REDIRECT_URL = '/accounts/profile/'
    LOGIN_URL = '/accounts/login/'
    LOGOUT_REDIRECT_URL = None
    MANAGERS = []
    MEDIA_ROOT = '/home/jose/.kolibri/media'
    MEDIA_URL = '/media/'
    MESSAGE_STORAGE = 'django.contrib.messages.storage.fallback.FallbackStorage'
    MIDDLEWARE = ['django.middleware.cache.UpdateCacheMiddleware', 'kolibri.core.analytics.middleware.MetricsMiddleware', 'django.contrib.sessions.middleware.SessionMiddleware', 'kolibri.core.device.middleware.KolibriLocaleMiddleware', 'django.middleware.common.CommonMiddleware', 'django.middleware.csrf.CsrfViewMiddleware', 'kolibri.core.auth.middleware.CustomAuthenticationMiddleware', 'django.contrib.messages.middleware.MessageMiddleware', 'django.middleware.clickjacking.XFrameOptionsMiddleware', 'django.middleware.security.SecurityMiddleware', 'django.middleware.cache.FetchFromCacheMiddleware', 'kolibri.core.webpack.middleware.WebpackErrorHandler']
    MIDDLEWARE_CLASSES = ['django.middleware.common.CommonMiddleware', 'django.middleware.csrf.CsrfViewMiddleware']
    MIGRATION_MODULES = {}
    MONTH_DAY_FORMAT = 'F j'
    NUMBER_GROUPING = 0
    PASSWORD_HASHERS = '********************'
    PASSWORD_RESET_TIMEOUT_DAYS = '********************'
    PREPEND_WWW = False
    REST_FRAMEWORK = {'UNAUTHENTICATED_USER': 'kolibri.core.auth.models.KolibriAnonymousUser', 'DEFAULT_RENDERER_CLASSES': ('rest_framework.renderers.JSONRenderer', 'rest_framework.renderers.BrowsableAPIRenderer'), 'EXCEPTION_HANDLER': 'kolibri.core.utils.exception_handler.custom_exception_handler'}
    ROOT_URLCONF = 'kolibri.deployment.default.dev_urls'
    SECRET_KEY = '********************'
    SECURE_BROWSER_XSS_FILTER = False
    SECURE_CONTENT_TYPE_NOSNIFF = False
    SECURE_HSTS_INCLUDE_SUBDOMAINS = False
    SECURE_HSTS_PRELOAD = False
    SECURE_HSTS_SECONDS = 0
    SECURE_PROXY_SSL_HEADER = None
    SECURE_REDIRECT_EXEMPT = []
    SECURE_SSL_HOST = None
    SECURE_SSL_REDIRECT = False
    SERVER_EMAIL = 'root@localhost'
    SESSION_CACHE_ALIAS = 'default'
    SESSION_COOKIE_AGE = 1200
    SESSION_COOKIE_DOMAIN = None
    SESSION_COOKIE_HTTPONLY = True
    SESSION_COOKIE_NAME = 'kolibri'
    SESSION_COOKIE_PATH = '/'
    SESSION_COOKIE_SECURE = False
    SESSION_ENGINE = 'django.contrib.sessions.backends.file'
    SESSION_EXPIRE_AT_BROWSER_CLOSE = True
    SESSION_FILE_PATH = '/home/jose/.kolibri/sessions'
    SESSION_SAVE_EVERY_REQUEST = False
    SESSION_SERIALIZER = 'django.contrib.sessions.serializers.JSONSerializer'
    SETTINGS_MODULE = 'kolibri.deployment.default.settings.dev'
    SHORT_DATETIME_FORMAT = 'm/d/Y P'
    SHORT_DATE_FORMAT = 'm/d/Y'
    SIGNING_BACKEND = 'django.core.signing.TimestampSigner'
    SILENCED_SYSTEM_CHECKS = ['auth.W004']
    STATICFILES_DIRS = []
    STATICFILES_FINDERS = ['django.contrib.staticfiles.finders.FileSystemFinder', 'django.contrib.staticfiles.finders.AppDirectoriesFinder']
    STATICFILES_STORAGE = 'django.contrib.staticfiles.storage.StaticFilesStorage'
    STATIC_ROOT = '/home/jose/.kolibri/static'
    STATIC_URL = '/static/'
    TEMPLATES = [{'BACKEND': 'django.template.backends.django.DjangoTemplates', 'DIRS': [], 'APP_DIRS': True, 'OPTIONS': {'context_processors': ['django.template.context_processors.debug', 'django.template.context_processors.request', 'django.contrib.auth.context_processors.auth', 'django.contrib.messages.context_processors.messages', 'kolibri.core.context_processors.custom_context_processor.developer_mode']}}]
    TEST_NON_SERIALIZED_APPS = []
    TEST_RUNNER = 'django.test.runner.DiscoverRunner'
    THOUSAND_SEPARATOR = ','
    TIME_FORMAT = 'P'
    TIME_INPUT_FORMATS = ['%H:%M:%S', '%H:%M:%S.%f', '%H:%M']
    TIME_ZONE = 'Europe/Madrid'
    USE_ETAGS = False
    USE_I18N = True
    USE_L10N = True
    USE_THOUSAND_SEPARATOR = False
    USE_TZ = True
    USE_X_FORWARDED_HOST = False
    USE_X_FORWARDED_PORT = False
    WSGI_APPLICATION = 'kolibri.deployment.default.wsgi.application'
    X_FRAME_OPTIONS = 'SAMEORIGIN'
    YEAR_MONTH_FORMAT = 'F Y'
    You're seeing this error because you have DEBUG = True in your
    Django settings file. Change that to False, and Django will
    display a standard page generated by the handler for this status code.







```

Reply:
Could you debug this inside of the `_user_is_admin_for_own_facility` function to see why that user is getting more than one `Facility` in that query? I'm not sure what the `Facility.dataset` field is and why it is not guaranteed to be unique to a facility.

Reply:
Yep the db had an incorrect FacilityDataset for one of the Facilities. Sorry for the noise